SolarSystem.java

```diff
@@ -0,0 +1,47 @@
+enum SolarSystem {
+    MERCURY(0, 0, 2439, null),
+    VENUS(1, 108, 6051, MERCURY),
+    EARTH(2, 149, 6371, VENUS),
+    MARS(3, 228, 3389, EARTH),
+    JUPITER(4, 778, 69911, MARS),
+    SATURN(5, 1427, 58232, JUPITER),
+    URANUS(6, 2871, 25362, SATURN),
+    NEPTUNE(7, 4497, 24622, URANUS);
+
+    private int distanceFromSun;
+    private int distanceFromPrevious;
+    private int radius;
+    private SolarSystem previousPlanet;
+    private SolarSystem nextPlanet;
+
+    SolarSystem(int prevDistance, int distanceFromSun, int radius, SolarSystem previous) {
```

Comment:
Task says that constructor must be different `SolarSystem(int prevDistance, int radius, SolarSystem previous)`. Calculate distanceFromSun in constructor